packages/envd/internal/services/filesystem/utils.go

```diff
@@ -15,3 +22,50 @@ func getEntryType(entry osEntry) rpc.FileType {
 		return rpc.FileType_FILE_TYPE_FILE
 	}
 }
+
+// getFileOwnership returns the owner and group names for a file.
+// If the lookup fails, it returns the numeric UID and GID as strings.
+func getFileOwnership(fileInfo os.FileInfo) (owner, group string) {
+	sys, ok := fileInfo.Sys().(*syscall.Stat_t)
+	if !ok {
+		return "", ""
+	}
+
+	// Look up username
+	if u, err := user.LookupId(fmt.Sprintf("%d", sys.Uid)); err == nil {
+		owner = u.Username
+	} else {
+		owner = fmt.Sprintf("%d", sys.Uid)
+	}
+
+	// Look up group name
+	if g, err := user.LookupGroupId(fmt.Sprintf("%d", sys.Gid)); err == nil {
+		group = g.Name
+	} else {
+		group = fmt.Sprintf("%d", sys.Gid)
+	}
+
+	return owner, group
+}
+
+func entryInfoFromFileInfo(fileInfo os.FileInfo, path string, sdkVersion string) *rpc.EntryInfo {
+	owner, group := getFileOwnership(fileInfo)
+	fileMode := fileInfo.Mode()
+
+	entry := &rpc.EntryInfo{
+		Name: fileInfo.Name(),
+		Type: getEntryType(fileInfo),
+		Path: path,
+	}
+
+	if sdkVersion != "" && sdkVersion >= "1.5.2" {
```

Comment:
why are we checking only version? The problem is python specific. Isn't the first check useless? Also `1.5.10` is less than `1.5.2` when compared as strings. Add comment with explanation